amp-pwa-reader/src/js/Article.js

```diff
@@ -41,6 +42,61 @@ class Article {
 
   }
 
+// see https://github.com/ampproject/amphtml/blob/master/spec/amp-shadow-doc.md
+// and https://jakearchibald.com/2016/fun-hacks-faster-content/
+// Recursively read and write chunks of data from the streaming API
+  stream() {
+    var shadowDoc = this.ampDoc;    // let's get this into the closure, and thus accessible to callback
+    var article = this;             // this too
+
+    fetch(this.proxyUrl).then(async response => {
+      let reader = response.body.getReader();
+      let decoder = new TextDecoder();
+
+      while (true) {
+        let chunk = await reader.read();
+
+        if (chunk.done) {
+          shadowDoc.writer.close();
+          break;
+        }
+
+        let html = decoder.decode(
+          chunk.value || new Uint8Array(),
+          {stream: !chunk.done}
+        );
+
+// check each chunk of HTML to see if it contains <style amp-custom>. If so, add in some extra CSS.
```

Comment:
can you indent all comments so they're on the right indentation as the block they're referring to?

Reply:
Yes - done!